logstash-core/lib/logstash/environment.rb

```diff
@@ -1,7 +1,39 @@
 # encoding: utf-8
 require "logstash/errors"
+require "logstash/config/cpu_core_strategy"
+require "logstash/settings"
 
 module LogStash
+
+  [
+            Setting::String.new("node.name", Socket.gethostname),
+            Setting::String.new("config.path", nil, false),
+            Setting::String.new("config.string", nil, false),
+           Setting::Boolean.new("config.test", false),
+           Setting::Boolean.new("config.reload.auto", false),
+           Setting::Numeric.new("config.reload.interval", 3),
+           Setting::Boolean.new("metric.collect", true) {|v| v == true }, # metric collection cannot be disabled
+            Setting::String.new("settings.dir", ::File.join(Environment::LOGSTASH_HOME, "config")),
```

Comment:
I'm ok with this. one question: what should we call the environment variable that sets the path for the directory containing logstash.yml ?
As a reminder, in logstash we use the notion of configuration for the pipeline description, so using LS_CONF_DIR can be misleading. Currently it's LS_SETTINGS_DIR